File: python/program/code/Taurus.py
"""
NAME: Taurus
AUTHOR: Tanaka Chitete
PURPOSE: Implement entry point for Taurus crypto-currency data analysis suite
CREATION: 03/03/2021
LAST MODIFICATION: 05/03/2021
"""

import InteractiveEntry
import ReportEntry
import sys

def main():
    # User provided incorrect number of CLAs for either mode
    if len(sys.argv) != 2 and len(sys.argv) != 4:
        print("Usage (Interactive): python Taurus.py -i")
        print("or")
        print("Usage (Report): python Taurus.py -r <asset filename> <trade filename>")
    else:
        mode = sys.argv[1]
        # User wants to run program in Interactive mode
        if mode == "-i":
            if len(sys.argv) != 2:
                print("Usage (Interactive): python3 Taurus -i")
            else:
                print() # Formatting
                InteractiveEntry.menu()
        # User wants to run program in Report mode
        elif mode == "-r":
            assetFilename = sys.argv[2]
            tradeFilename = sys.argv[3]
            if len(sys.argv[1]) != 4:
                print("Usage (Report): python3 Taurus.py -r <asset filename> <trade filename>")
            else:
                print() # Formatting
                ReportEntry.entry(assetFilename, tradeFilename)
        # User hasn't specified that they want to run program in either Interactive or Report
        else:
            print("Usage (Interactive): python Taurus.py -i")
            print("or")
            print("Usage (Report): python Taurus.py -r <asset filename> <trade filename>")            


main()

File: java/Taurus.java
/*
 * NAME: Taurus
 * CREATOR: Tanaka Chitete
 * STUDENT_ID: 20169321
 * UNIT: COMP1002
 * PURPOSE: Provide entry point for crypto trading data analysis suite
 * CREATION: 30/09/2020
 * LAST MODIFICATION: 06/11/2020
 */

public class Taurus {
    public static final int INTERACTIVE_ARG_COUNT = 1;
    public static final int REPORT_ARG_COUNT = 3;

    public static final String INTERACTIVE = "-i";
    public static final String REPORT = "-r";
    
    public static void main (String[] args) {
        String mode;
        String assetFilename;
        String tradeFilename;

        // User has not specified correct number of CLAs for either mode
        int argc = args.length;
        if (argc != INTERACTIVE_ARG_COUNT && argc != REPORT_ARG_COUNT) {
            // Informs user of correct Interactive and Report usage
            System.out.println("Usage (Interactive): java -cp .:json-20200518.jar Taurus -i");
            System.out.println("or");
            System.out.println("Usage (Report):      java -cp .:json-20200518.jar Taurus -r " + 
                "<asset filename> <trade filename>");
        }
        else {
            mode = args[0];
            // User specifies Interactive
            if (mode.equals(INTERACTIVE)) {
                mode = args[0];
                // User has provided incorrect number of CLAs for Interactive
                if (argc != INTERACTIVE_ARG_COUNT) {
                    // Informs user of correct Interactive usage
                    System.out.println("Usage (Interactive): java -cp .:json-20200518.jar Taurus " +
                        "-i");
                }
                // Runs in Interactive
                else {
                    System.out.println();
                    InteractiveEntry.entry();
                }
            }
            // User specifies Report
            else if (mode.equals(REPORT)) {
                mode = args[0];
                assetFilename = args[1];
                tradeFilename = args[2];
                // User has provided incorrect number of CLAs for Report
                if (argc != REPORT_ARG_COUNT) {
                    // Informs user of correct report usage
                    System.out.println("Usage (Report): java -cp .:json-20200518.jar Taurus -r " + 
                        "<asset filename> <trade filename>");
                }
                // Runs in Report
                else {
                    System.out.println();
                    ReportEntry.entry(assetFilename, tradeFilename);
                }
            }
            // User specified neither Interactive nor Report
            else {
                // Informs user of correct mode usages
                System.out.println("Usage (Interactive): java -cp .:json-20200518.jar Taurus -i");
                System.out.println("or");
                System.out.println("Usage (Report):      java -cp .:json-20200518.jar Taurus -r " + 
                    "<asset filename> <trade filename>");
            }    
        }
    }
}
